core/Segment.php

```diff
@@ -586,6 +608,10 @@ public static function getSegmentHash($definition)
      */
     public function getSelectQuery($select, $from, $where = false, $bind = array(), $orderBy = false, $groupBy = false, $limit = 0, $offset = 0, $forceGroupBy = false)
     {
+        if (Development::isEnabled() && !empty($this->missingDatesException)) {
```

Comment:
Hm, it does make sense to not show the warning whenever a segment is created but then also not used.

However, would it also make sense to include a double stacktrace? One for "segment created at" and one for "segment used at"? Depending on the code it may be difficult to find out where it was used, if you only get the information where it was created, unless you know how to reliably reproduce the error and get a breakpoint triggered :thinking: